src/com/powem/inv/algos/ConfigParser.java

```diff
@@ -0,0 +1,143 @@
+package com.powem.inv.algos;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+//        Problem: Configuration Parser
+//        Problem Statement:
+//        Develop a parser for a configuration file format. The file contains key-value pairs, where each key is
+//        separated from its value by an equals sign (=). Comments in the file start with a hash symbol (#) and should
+//        be ignored.
+//
+//        Create a parser that reads a configuration string, extracts key-value pairs, and ignores any lines that are
+//        comments or improperly formatted.
+//
+//        Keys and values are strictly alphanumeric and underscored (no spaces).
+//        Each valid key-value pair appears on a new line.
+//        Lines that do not conform to the key-value format should be ignored.
+//
+//        Implement the ConfigParser class that takes a configuration string and provides methods to get the value for
+//        a specified key and list all keys.
+
+//        Function signature:
+//        public class ConfigParser {
+//          private Map<String, String> configMap;
+//        }
+//
+
+public class ConfigParser {
+    private Map<String, String> configMap;
+
+    public ConfigParser(String config) {
+        this.configMap = parseConfig(config);
+    }
+
+    public String getValue(String key) {
+        return configMap.get(key);
+    }
+
+    public Set<String> getAllKeys() {
+        return configMap.keySet();
+    }
+
+    private Map<String, String> parseConfig(String config) {
```

Comment:
this problem is too basic. there's no complex data structures or algorithms used